DB/src/main/java/io/deephaven/db/tables/utils/TableManagementTools.java

```diff
@@ -137,7 +137,9 @@ private static Table readParquetTable(@NotNull final File source, final boolean
         ParquetInstructions readInstructions = ParquetInstructions.EMPTY;
         try {
             final String path = source.getPath() + ((!isDirectory) ? "" : File.separator + ParquetTableWriter.PARQUET_FILE_NAME);
-            readInstructions = ParquetReaderUtil.readParquetSchema(path, readInstructions, colConsumer);
+            readInstructions = ParquetReaderUtil.readParquetSchema(path, readInstructions, colConsumer,
+                    (final String colName) -> DBNameValidator.legalizeColumnName(colName, s -> s.replace(" ", "_")));
```

Comment:
Nice! If you wanted to, you could sort of go one step further in all this by also passing the keySet of DH column names into the legalize method as a third arg; that appears to actually help you avoid duplication, and doesn't try to mutate the set.